sdk/python/kfp/dsl/pipeline_channel.py

```diff
@@ -24,17 +24,20 @@
 
 
 @dataclasses.dataclass
-class ConditionOperator:
-    """Represents a condition expression to be used in dsl.Condition().
+class BinaryOperation:
```

Comment:
`negate` is a unary operator, so the name here seems not accurate.

Reply:
I've been thinking of this class as a binary operation comprised of an `operator` between a `left_operand` and a `right_operand`, with the `negate` flag as a secondary consideration on top of the `BinaryOperation` instance's core data (the other three fields). It seems a bit too heavyweight to split out these abstractions entirely into `BinaryOperation` and `UnaryOperation`/`Negation` classes. I'd prefer to either (a) keep the name as-is, (b) rename to something like `BinaryOperationWithNegation`, or (c) rename to `Operation`. I think I (weakly) prefer those options in that order. Do you have a preference?

Reply:
I agree not to split the class into two. But the naming is still a bit of as I read the class name I won't think of `negate` being part of it. What do you think about `ConditionOperation`?

Reply:
SG. Updated.